### PR TITLE
Page for Posts: Display notice in template panel

### LINKED ISSUE #38487: When editing the Posts Page, the Template selector doesn't work
To reproduce:

1. Set a static homepage
2. Set a page to use as the "Posts page"
3. Edit the "Posts page"
4. Change the template in the "Template" panel of the document inspector to something custom (create a new template if necessary)
5. Save
6. Notice that the chosen template is not used

To be clear, I think it is appropriate that the template for this page cannot be changed, as that would allow folks to "break" the Posts page. However the appearance of the Template panel suggests that it _can_ be changed, and the subsequent experience feels broken.

### PR DESCRIPTION
**Pull request by Mamaduka**
## Description
PR displays notice inside the "Templates" panel when editing "Page for Posts." Changing the template for this page has no effects; ~~this also [matches logic](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/meta-boxes.php#L976) in Classic Editor.~~

## Testing Instructions
1. Using Settings -> Reading, set static page.
2. Set a post page.
3. Edit this page and check that the Templates panel isn't displayed.
4. Edit another page and confirm that the Templates panel is visible.

## Screenshots 
CleanShot 2022-02-08 at 11 19 31

## Types of changes
Bugfix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

File: lib/full-site-editing/edit-site-page.php [hunk 1]
```diff
@@ -223,6 +223,16 @@ function register_site_editor_homepage_settings() {
...
+
+	register_setting(
+		'reading',
+		'page_for_posts',
+		array(
+			'show_in_rest' => true,
+			'type'         => 'number',
+			'description'  => __( 'The ID of the page that should display the latest posts', 'gutenberg' ),
+		)
+	);
```
Comment (Mamaduka) [inline lib/full-site-editing/edit-site-page.php:235]: I noticed that static page settings aren't available when the plugin is deactivated. I think we forgot to expose these settings in REST API for 5.9.

**Comment by jameskoster** [comment]
How do you feel about keeping the panel, but displaying a message explaining why the template cannot be changed?

If we remove the entire panel we also remove the shortcut to edit the current template which seems a bit of a shame.

**Comment by Mamaduka** [comment]
@jameskoster, I've updated the logic. Let me know if you want to adjust the design.

CleanShot 2022-02-08 at 14 50 37

**Comment by jameskoster** [comment]
Looks like there's a small typo there "connot", otherwise it's good. 

The "Edit" link looks a bit lost, but we'll most likely address that as a part of https://github.com/WordPress/gutenberg/issues/31591 anyway.

**Comment by Mamaduka** [comment]
> Looks like there's a small typo there "connot", otherwise it's good.

Thanks for catching this.

**Comment by jameskoster** [comment]
Nice. Do you think we should remove the dropdown in the quick edit interface as well, or is that something to do separately?

**Comment by Mamaduka** [comment]
> Do you think we should remove the dropdown in the quick edit interface as well, or is that something to do separately?

That probably needs to be fixed in WP core.

**Review by jameskoster — APPROVED** [review]

**Comment by Mamaduka** [comment]
I think settings from `register_site_editor_homepage_settings` haven't been ported to WP 6.0, and this fix doesn't work as expected.

I'm tentatively adding this PR to the 6.0.1 board.

cc @gziolo, @adamziel

**Comment by gziolo** [comment]
Good catch @Mamaduka. For the person that will work on the fix, we are talking about:

Screenshot 2022-05-26 at 15 44 03

that should be added directly to the settings endpoint.

**Comment by adamziel** [comment]
@Mamaduka would you mind preparing a backport against the 6.0 branch of the wordpress-develop repo so that this could be backported to 6.01 Beta 1 on Tuesday?

**Comment by adamziel** [comment]
This PR is now backported to 6.0 branch so I'm removing the label.

**Pull request merged**